odoo/modules/module.py

```diff
@@ -160,8 +160,10 @@ def initialize_sys_path() -> None:
     sys.modules["odoo.addons.base.maintenance"] = maintenance_pkg
     sys.modules["odoo.addons.base.maintenance.migrations"] = odoo.upgrade
 
-    # hook deprecated module alias from openerp to odoo and "crm"-like to odoo.addons
+    # hook for upgrades and namespace freeze
     if not getattr(initialize_sys_path, 'called', False):  # only initialize once
+        odoo.addons.__path__._path_finder = lambda *a: None  # prevent path invalidation
+        odoo.upgrade.__path__._path_finder = lambda *a: None  # prevent path invalidation
```

Comment:
annoying two lines

Reply:
Required because we monkeypatch the `__path__` of the namespace and during invalidation it rebuilds it based on the parent namespace. Removing the path finder prevents invalidation.